@here/harp-mapview/lib/text/TextElementsRenderer.ts

```diff
@@ -1687,7 +1692,12 @@ export class TextElementsRenderer {
             const textNeedsDraw = !textRejected || textRenderState!.isFading();
 
             if (textNeedsDraw) {
-                if (pointLabel.textReservesSpace) {
+                // Don't allocate space for rejected text. When zooming, this allows placement of a
+                // lower priority text element that was displaced by a higher priority one (not
+                // present in the new zoom level) before an even lower priority one takes the space.
+                // Otherwise the lowest priority text will fade in and back out.
+                // TODO: Add a unit test for this scenario.
+                if (pointLabel.textReservesSpace && !textRejected) {
```

Comment:
is this related?

Reply:
This is not related to your other comment above. This is a minor improvement that avoids flickering in particular scenario:
Imagine we are look at "EUROPE" label, then we zoom in to a level where continent labels are disabled and capitals become visible. "London" starts fading in to the west side (it doesn't collide with "EUROPE") as "EUROPE" keeps fading out. As soon as "EUROPE" becomes invisible, the label "DEUTSCHLAND" that was in collision both with "EUROPE" and "London" starts fading in, since it has less priority than "EUROPE" but more than "London", and causes the rejection of "London" that starts fading out.
If we let "DEUTSCHLAND" fade in as soon as "EUROPE" starts fading out, then "London" will never fade in in the first place.

Reply:
Ok, please move to a separate PR, if there is any regression, it would be easier to revert this single line than have to revert everything.

Reply:
I've moved it to a separate commit, so that it could reverted if necessary as you say.
But I left it in the same PR, i won't squash it when merging. The change is related to zoom level persistence since the scenario I described is reproducible as soon as zoom level persistence is introduced.

Reply:
good idea, yes it can be a separate commit, cool!

Reply:
wait, separate commit is cool, but please merge the other two commits.

Reply:
will do